src/components/Search/utils.ts

```diff
@@ -27,161 +25,249 @@ export function updateQueryStringParameter(
 
-    nestedQuery = {
-      must: [
-        {
-          bool: {
-            should: [
-              {
-                query_string: {
-                  query: `${modifiedSearchTerm}`,
-                  fields: searchFields,
-                  minimum_should_match: '2<75%',
-                  phrase_slop: 2,
-                  boost: 5
-                }
-              },
-              {
-                query_string: {
-                  query: `${noSpaceSearchTerm}*`,
-                  fields: searchFields,
-                  boost: 5,
-                  lenient: true
-                }
-              },
-              {
-                match_phrase: {
-                  content: {
-                    query: `${searchTerm}`,
-                    boost: 10
-                  }
-                }
-              },
-              {
-                query_string: {
-                  query: `${prefixedSearchTerm}`,
-                  fields: searchFields,
-                  default_operator: 'AND'
+  const prefixedSearchTerm =
+    emptySearchTerm && searchTerm
+      ? searchTerm
+      : !emptySearchTerm && searchTerm
+      ? '*' + searchTerm + '*'
+      : '**'
+  const searchFields = [
+    'id',
+    'nft.owner',
+    'datatokens.address',
+    'datatokens.name',
+    'datatokens.symbol',
+    'metadata.name^10',
+    'metadata.author',
+    'metadata.description',
+    'metadata.tags'
+  ]
+
+  const nestedQuery = {
+    must: [
+      {
+        bool: {
+          should: [
+            {
+              query_string: {
+                query: `${modifiedSearchTerm}`,
+                fields: searchFields,
+                minimum_should_match: '2<75%',
+                phrase_slop: 2,
+                boost: 5
+              }
+            },
+            {
+              query_string: {
+                query: `${noSpaceSearchTerm}*`,
+                fields: searchFields,
+                boost: 5,
+                lenient: true
+              }
+            },
+            {
+              match_phrase: {
+                content: {
+                  query: `${searchTerm}`,
+                  boost: 10
                 }
               }
-            ]
-          }
+            },
+            {
+              query_string: {
+                query: `${prefixedSearchTerm}`,
+                fields: searchFields,
+                default_operator: 'AND'
+              }
+            }
+          ]
         }
-      ]
-    }
+      }
+    ]
   }
-  accessType && filters.push(getFilterTerm('services.type', accessType))
-  serviceType && filters.push(getFilterTerm('metadata.type', serviceType))
-  complianceType &&
-    filters.push(
-      getFilterTerm(
-        'metadata.additionalInformation.compliance.keyword',
-        complianceType
-      )
-    )
+  filters &&
+    filters.forEach((filter) => {
+      if (filterTerms.has(filter.category)) {
+        if (filter?.term) {
```

Comment:
you mentioned it can be both a term and range, but this code does not really reflect that. just noting that here

Reply:
filterTerms is a `Map<string, (FilterTerm | FilterRange)[]>` because the range queries will be placed using the same AND and OR logic that we added to the query yesterday. However the structure of a range query is different from a term query and it needs to be parsed separately.